src/prefect/input/run_input.py

```diff
@@ -280,6 +280,60 @@ def receive(
             flow_run_id=flow_run_id,
         )
 
+    @classmethod
+    def subclass_from_base_model_type(
+        cls, model_cls: Type[pydantic.BaseModel]
+    ) -> Type["RunInput"]:
+        """
+        Create a new `RunInput` subclass from the given `pydantic.BaseModel`
+        subclass.
+
+        Args:
+            - model_cls (pydantic.BaseModel subclass): the class from which
+                to create the new `RunInput` subclass
+        """
+        return type(f"{model_cls.__name__}RunInput", (RunInput, model_cls), {})  # type: ignore
+
+
+class AutomaticRunInput(RunInput):
+    value: Any
```

Comment:
You could assign this a typevar (like we use `T`) and then you could load's return type be that typevar, I _think_ that would give nice type handling, so if a user were to pass `int` through it should come back out as `int` in their code. You might also need to update `AutomaticRunInput` to be `AutomaticRunInput[V]` in `subclass_from_type` and make this class a `Generic[V]`. If that all sounds messy I can also just play with it after this is merged and get it working.